glommio/benches/echo.rs

```diff
@@ -0,0 +1,150 @@
+use std::time::Instant;
+
+use futures_lite::{
+    future::block_on,
+    io::{copy, split},
+};
+use glommio::{net::TcpListener, prelude::*};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    runtime::{Builder, Handle, Runtime},
+    sync::oneshot,
+};
+
+#[derive(Copy, Clone)]
+struct TestCfg {
+    clients: usize,
+    requests: usize,
+    msg_size: usize,
+    short_lived: bool,
+}
+
+impl TestCfg {
+    fn new(clients: usize, requests: usize, msg_size: usize, short_lived: bool) -> Self {
+        Self {
+            clients,
+            requests,
+            msg_size,
+            short_lived,
+        }
+    }
+}
+
+fn main() {
+    let glommio_server_port: u16 = 9100;
+    let tokio_server_port: u16 = 9200;
+
+    let runtime = Builder::new_multi_thread()
+        // client runs in worker thread
+        .worker_threads(1)
+        // tokio server runs in blocking thread
+        .max_blocking_threads(1)
+        .enable_io()
+        .build()
+        .unwrap();
+
+    start_glommio_server(glommio_server_port);
+    start_tokio_server(&runtime, tokio_server_port);
+
+    let mut test_cfgs = vec![];
+    for sz in [128, 512, 1000] {
+        for c in [1, 50, 150, 300, 500, 1000] {
+            test_cfgs.push(TestCfg::new(c, 1_000_000, sz, false));
+        }
+    }
+
+    for cfg in test_cfgs {
+        let qps = start_clients(&runtime, "Tokio", tokio_server_port, cfg, None);
+        start_clients(&runtime, "Glommio", glommio_server_port, cfg, Some(qps));
+        println!("--")
+    }
+    runtime.shutdown_background();
+}
+
+fn start_glommio_server(port: u16) {
+    let (ready_send, ready_recv) = oneshot::channel();
+
+    LocalExecutorBuilder::default()
+        .spawn(move || async move {
+            let listener = TcpListener::bind(("0.0.0.0", port)).unwrap();
+            ready_send.send(()).unwrap();
+
+            while let Ok(stream) = listener.accept().await {
+                spawn_local(async move {
+                    let (inbound, outbound) = split(stream);
+                    copy(inbound, outbound).await.ok();
+                })
+                .detach();
+            }
+        })
+        .unwrap();
+
+    block_on(ready_recv).unwrap();
+}
+
+fn start_tokio_server(runtime: &Runtime, port: u16) {
+    use tokio::{io::copy, net::TcpListener};
+
+    let (ready_send, ready_recv) = oneshot::channel();
+
+    runtime.spawn_blocking(move || {
```

Comment:
It is used to ensure the tokio servers will not use more than one cpu's resource, or it could be unfair to its glommio competitior. It would be more straightforward to create separate runtimes for client and server, though.

Reply:
I just don't have a good idea of the implications, personally.. It just struck me as suspicious in some way. It could be fine, I just don't know if it will taint the output.